File: taxifleet_backend/src/main/java/ru/taxifleet/entity/Order.java
package ru.taxifleet.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import ru.taxifleet.enums.OrderStatus;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "orders")
@Getter @Setter @NoArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "address_from", nullable = false, length = 255)
    private String addressFrom;

    @Column(name = "address_to", nullable = false, length = 255)
    private String addressTo;

    @Column(name = "client_name", nullable = false, length = 100)
    private String clientName;

    @Column(name = "client_phone", nullable = false, length = 20)
    private String clientPhone;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private OrderStatus status = OrderStatus.NEW;

    @Column(precision = 10, scale = 2)
    private BigDecimal cost;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "admin_id")
    private Admin admin;

    @PrePersist
    public void prePersist() {
        this.createdAt = LocalDateTime.now();
        if (this.status == null) {
            this.status = OrderStatus.NEW;
        }
    }

    public boolean isNew() {
        return this.status == OrderStatus.NEW;
    }

    public void cancel() {
        this.status = OrderStatus.CANCELLED;
    }

    public void complete() {
        this.status = OrderStatus.DONE;
    }
}

File: taxifleet_app/lib/models/order.dart
class Order {
  final int id;
  final String addressFrom;
  final String addressTo;
  final String clientName;
  final String clientPhone;
  final String status;
  final double? cost;
  final String createdAt;

  Order({
    required this.id,
    required this.addressFrom,
    required this.addressTo,
    required this.clientName,
    required this.clientPhone,
    required this.status,
    this.cost,
    required this.createdAt,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['id'] as int,
      addressFrom: json['addressFrom'] as String? ?? '',
      addressTo: json['addressTo'] as String? ?? '',
      clientName: json['clientName'] as String? ?? '',
      clientPhone: json['clientPhone'] as String? ?? '',
      status: json['status'] as String? ?? 'NEW',
      cost: (json['cost'] as num?)?.toDouble(),
      createdAt: json['createdAt'] as String? ?? '',
    );
  }
}
